### ISSUE TITLE
Problem in the Contributing Flow

### ISSUE THREAD

**Issue by WesleyBatista** [datasciencebr/serenata-de-amor#288]
Hello guys!

**TLDR**
Maybe we should have an alternative guide to install the dev environment. 

---

My first time here since I heard about the project last year. Sorry in advance if I'm doing something in the wrong way.

Following the [CONTRIBUTING.md](https://github.com/datasciencebr/serenata-de-amor/blob/master/CONTRIBUTING.md) guide I jumped to the post on [medium](https://medium.com/data-science-brigade/how-does-one-contribute-to-serenata-de-amor-operation-36e3e7b38207#3abf), which in turn says that I must install the library using the Git URL. So I did that using my fresh new fork and I got what follows:
```
$ pip install -e git://github.com/WesleyBatista/serenata-de-amor.git@master#egg=Serenata                                                                                                 1 ↵
Obtaining Serenata from git+git://github.com/WesleyBatista/serenata-de-amor.git@master#egg=Serenata
  Cloning git://github.com/WesleyBatista/serenata-de-amor.git (to master) to ./src/serenata
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
    IOError: [Errno 2] No such file or directory: '/home/wesley/github/projects/serenata-de-amor/src/serenata/setup.py'
    
    ----------------------------------------
```

It failed trying to use the *setup.py* file, which doesn't exists. But, in place, we have a executable file called just **setup**.

To get rid from the issue I've just ignored all and tried to install by myself. I can resume the steps as follows:

1. Install virtualenv: `pip install --user virtualenv`
2. Create a virtual env using python 3: `virtualenv -p python3 .venv_python3`
3. Use the virtual env that we've just created `source .venv_python3/bin/activate`
4. Run the executable `./setup`

>It failed due to some **matplotlib** dependencies twice. So I needed to install also those two:
>
>  1. `sudo apt-get install libfreetype6-dev`
>  2. `sudo apt-get install python3.5-dev`

5. Run `./setup` again

It worked! :smile:  🎉🎉🎉
```
Downloading datasets (this might take several minutes     depending on your internet connection)
Downloading 32 files: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 244M/244M [02:30<00:00, 1.30Mb/s]
```

It all started with the "setup" file thing.  But I had also had problems related to the environment. So we kind of have two issues here.

Maybe I had these problems also due to the fact that I won't install Anaconda. It is just weird for me, and mess with my work environment. 

But that's it! If someone have this problem too, at least we have here something that can possibly help.

*I'm using here Ubuntu 16.04*

**Comment by cuducos** [comment]
Just to be clear… did you use Anaconda Python as recommended?

**Comment by WesleyBatista** [comment]
No, I did not use Anaconda Python.

**Comment by cuducos** [comment]
Hi @WesleyBatista,

Ok… surely the issue has shown our contributing guide is indeed confusing. The role of the medium article was merely to introduce what the repos are about (this one, Rosie, Jarbas, the toolbox). The core of the instructions to [get a working dev environment is still at the contribution guide](https://github.com/datasciencebr/serenata-de-amor/blob/master/CONTRIBUTING.md#local-installation-environment-without-docker).

There are a couple of differences between what you've tried and what is documented there: mainly we start from Anaconda Python and then we opt for a conda env instead of a virtualenv. This is **not** necessarily — if you wanna avoid Anaconda you can workaround as you did, but this way is too cumbersome to be documented in a way that is platform independent, package manager independent and so on.

So… basically part of the errors you reported are expected _if you don't user Anaconda Python_ as recommended. What else do you think could be better documented? How could using this Anaconda Python, conda env and running the `setup` could be better communicated?

**Comment by WesleyBatista** [comment]
After a quick talk with a contributor here on #PyBR13, I guess that now I know what happened.

I've started on **serenata-de-amor** project, but the `pip install` using GIT URL information was about the **serenata-toolbox** repo. In the later one repo, there is a **setup.py** in the root directory.

My bad, sorry :/

In response to your question, I think that maybe we could rethink about the **Before you start** section, or the part related to the article...
It happened because I've jumped from CONTRIBUTING.md to the article. 

But I don't really know, I am just starting here. Also, my focus was actually on this another #275 :)

Maybe in the future I will have a wider understanding about the components, and then I will be able to better contribute on that. 

For now, you can close the issue if you want.

**Comment by cuducos** [comment]
> But I don't really know, I am just starting here.

The point of view is extremely important for us! Having feedback from starters as you is the only way to ponder on our biased look into the repo ; )

Keep this stuff in mind. And many thanks for the feedback.
